File: lib/src/main/java/com/sd/lib/json/FJson.java
package com.sd.lib.json;

import com.squareup.moshi.JsonAdapter;
import com.squareup.moshi.Moshi;
import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory;

public final class FJson {

    public static final Moshi Moshi = new Moshi.Builder()
            .add(new KotlinJsonAdapterFactory())
            .build();

    private FJson() {
    }

    public static <T> T fromJson(String json, Class<T> clazz) throws Exception {
        final JsonAdapter<T> adapter = Moshi.adapter(clazz);
        return adapter.fromJson(json);
    }

    public static <T> String toJson(T instance, Class<T> clazz) {
        final JsonAdapter<T> adapter = Moshi.adapter(clazz);
        return adapter.toJson(instance);
    }
}

File: lib/src/main/java/com/sd/lib/json/FJson.kt
package com.sd.lib.json

import com.squareup.moshi.Moshi

object FJson {

    @JvmStatic
    val moshi: Moshi = Moshi.Builder().addAdapterIgnoreNull().build()

    /**
     * 忽略json中值为null的字段
     */
    fun addAdapterIgnoreNull(builder: Moshi.Builder) {
        builder.addLast(StandardJsonAdapters.FACTORY)
        builder.addLast(FKotlinJsonAdapterFactory())
    }

    @JvmStatic
    fun <T> fromJson(json: String?, clazz: Class<T>): T? {
        if (json.isNullOrEmpty()) return null
        return moshi.adapter(clazz).fromJson(json)
    }

    @JvmStatic
    fun <T> toJson(instance: T, clazz: Class<T>): String {
        return moshi.adapter(clazz).toJson(instance)
    }
}

/**
 * 忽略json中值为null的字段
 */
fun Moshi.Builder.addAdapterIgnoreNull(): Moshi.Builder {
    FJson.addAdapterIgnoreNull(this)
    return this
}